### PR TITLE
BUG: Fix interpolate ValueError for datetime64_tz index

### LINKED ISSUE #27548: pd.Series.interpolate() ValueError for some methods with TZ localized index
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

ser = pd.Series([1.0, pd.np.nan, 2.0], index=pd.date_range(
    start='20190101T0000', freq='1h', periods=3))
print(ser.interpolate(method='pad'))
print(ser.tz_localize('UTC').interpolate(method='pad'))  # raises ValueError
```
#### Problem description

Interpolation of a Series with a TZ localized DatetimeIndex raises a ValueError when method  is not one of (linear, index, values, nearest, time). This does not occur with version 0.24.2, and I think the bug was introduced in https://github.com/pandas-dev/pandas/pull/25394.

#### Expected Output
2019-01-01 00:00:00    1.0
2019-01-01 01:00:00    1.0
2019-01-01 02:00:00    2.0
Freq: H, dtype: float64
2019-01-01 00:00:00+00:00    1.0
2019-01-01 01:00:00+00:00    1.0
2019-01-01 02:00:00+00:00    2.0
Freq: H, dtype: float64

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
                                            
INSTALLED VERSIONS     
------------------      
commit           : d749aae9549329912edabc646fcda4083af18e22
python           : 3.7.2.final.0
python-bits      : 64  
OS               : Linux
OS-release       : 5.1.12-300.fc30.x86_64
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None 
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.0+19.gd749aae954
numpy            : 1.16.4
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 18.1
setuptools       : 40.6.2
Cython           : 0.29.12
pytest           : 5.0.1
hypothesis       : 4.28.2
sphinx           : 1.8.5
blosc            : 1.8.0
feather          : None
xlsxwriter       : 1.1.8
lxml.etree       : 4.3.4
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.6.1
pandas_datareader: None
bs4              : 4.8.0
bottleneck       : 1.2.1
fastparquet      : 0.3.1
gcsfs            : None
lxml.etree       : 4.3.4
matplotlib       : 3.1.1
numexpr          : 2.6.9
odfpy            : None
openpyxl         : 2.6.2
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : 0.3.0
scipy            : 1.3.0
sqlalchemy       : 1.3.6
tables           : 3.5.2
xarray           : 0.12.3
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.1.8

</details>

### PR DESCRIPTION
**Pull request by alorenzo175**
- [ x ] closes #27548 
- [ x ] tests added / passed
- [ x ] passes `black pandas`
- [ x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ x ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.25.1.rst [hunk 1]
```diff
@@ -56,7 +56,7 @@ Timezones
 
 Numeric
 ^^^^^^^
--
+- Bug in :meth:`interpolate` when using a timezone aware index (:issue:`27548`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v0.25.1.rst]: ```... when using a timezone aware :class:`DatetimeIndex` ...```

File: doc/source/whatsnew/v0.25.1.rst [hunk 2]
```diff
@@ -56,7 +56,7 @@ Timezones
 
 Numeric
 ^^^^^^^
--
+- Bug in :meth:`interpolate` when using a timezone aware :class:`DatetimeIndex` (:issue:`27548`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.1.rst]: this won't render, you want ```:meth:`Series.interpolate` ```

**Comment by jreback** [comment]
looks ok, ping when changed & green.

File: pandas/tests/series/test_missing.py [hunk 3]
```diff
@@ -1521,12 +1521,23 @@ def test_interp_nonmono_raise(self):
             s.interpolate(method="krogh")
 
     @td.skip_if_no_scipy
-    def test_interp_datetime64(self):
+    @pytest.mark.parametrize("method", ["nearest", "pad"])
```
Comment (jreback) [inline pandas/tests/series/test_missing.py]: you can parametrize on tz_naive_fixture here (and just remove the bottom test); this includes None and a bunch of timezones.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by alorenzo175** [comment]
@jreback I've updated to use the tz_naive_fixture and checks are green

**Review by mroeschke — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @alorenzo175

**Pull request merged**